sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/GridUtils.groovy

```diff
@@ -0,0 +1,93 @@
+        } catch (IllegalStateException illegalStateException) {
+            log.error("Provided component could not be made responsive . $illegalStateException.message")
+            log.debug("Provided component does not support being set to responsive. $illegalStateException.message", illegalStateException)
+        }
+    }
+
+    /**
+     * Adjusts the width of the grid and it's associated hotbar layout when the grid is manually resized by the user
+     *
+     * @param AbstractComponent hotbarLayout the {@link com.vaadin.ui.AbstractComponent}
+     * @param Grid grid the {@link com.vaadin.ui.Grid},
+     *
+     * @since 1.0.2
+     */
+    static void makeGridResponsiveToResize(Grid grid, AbstractComponent hotbarLayout) {
+
+        //The grid width should adjust to the resizing preferences of the user
+        try {
+            grid.addColumnResizeListener(columnListener -> {
+                int maximumPortletWidth = computeMaximumLiferayPortletWidth()
+                double columnsWidth = 0
+                grid.getColumns().each { column ->
+                    columnsWidth += column.getWidth()
+                }
+                if (columnsWidth <= maximumPortletWidth) {
+                    grid.setWidth(Math.floor(columnsWidth).toString())
+
+                } else {
+                    //If projectGrid width is bigger than screen adjust grid to max screen size
+                    grid.setWidth(maximumPortletWidth.toString())
+                }
+                if (hotbarLayout) {
+                    if (columnsWidth <= maximumPortletWidth) {
+                        hotbarLayout.setWidth(Math.floor(columnsWidth).toString())
+                    } else {
+                        hotbarLayout.setWidth(maximumPortletWidth.toString())
+                    }
+                }
+            })
+        } catch (Exception exception) {
+            log.error("An error occured during manual grid resizing. $exception.message")
+            log.debug("An error occured during manual grid resizing. $exception.message", exception)
+        }
+    }
+
+    /**
+     * Computes the maximum width a component can have in the liferay portal
+     * @since 1.0.2
+     * @returns maximumPortletWidth maximum width a portlet can take accounting for the liferay margins and paddings
+     */
+
+    private static int computeMaximumLiferayPortletWidth() {
+        try {
+            int browserWindowWidth = UI.getCurrent().getPage().getBrowserWindowWidth()
+            //We need to adjust the grid width to the liferay padding
+            int maximumPortletWidth = Math.floor(0.88 * browserWindowWidth).toInteger()
+            return maximumPortletWidth
+        } catch (Exception exception) {
+            log.error("Unable to compute browser page width $exception.message")
+            log.debug("An error occured during the computation of maximum possible portlet width $exception.message", exception)
+        }
+    }
```

Comment:
I don't think we should make our code dependent on Liferay here. (or even its configuration)
By hardcoding the 88% here, the whole project is dependent on the liferay configuration.